server/chat-plugins/help.ts

```diff
@@ -0,0 +1,469 @@
+			const force = cmd === 'forceadd';
+			if (force && (!REGEX_WHITELIST.includes(user.id) && !user.can('rangeban'))) {
+				return this.errorReply(`You cannot use raw regex - use /helpfilter add instead.`);
+			}
+			if (!this.can('ban', null, helpRoom)) return false;
+			Answerer.tryAddRegex(target, force);
+			this.privateModAction(`${user.name} added regex for "${target.split('=>')[0]}" to the filter.`);
+			this.modlog(`HELPFILTER ADD`, null, target);
+		},
+		remove(target, room, user) {
+			if (!helpRoom) return this.errorReply(`No Help room to manage the filter for.`);
+			if (!this.can('ban', null, helpRoom)) return false;
+			const [faq, index] = target.split(',');
+			// intended for use mainly within the page, so supports being used in all rooms
+			this.room = helpRoom;
+			const num = parseInt(index);
+			if (isNaN(num)) return this.errorReply("Invalid index.");
+			Answerer.tryRemoveRegex(faq, num - 1);
+			this.privateModAction(`${user.name} removed regex ${num} from the usable regexes for ${faq}.`);
+			this.modlog('HELPFILTER REMOVE', null, index);
+		},
+		queue(target, room, user) {
+			if (room?.roomid !== 'help') return this.errorReply(`This command is only available in the Help room.`);
+			if (!helpRoom) return this.errorReply(`No Help room to manage the filter for.`);
+			if (!this.can('show', null, helpRoom)) return false;
+			if (!room.auth.has(user.id)) return this.errorReply(`Only roomauth can submit regexes to the filter.`);
+			if (!target) return this.errorReply(`Specify regex.`);
+			const faq = Answerer.getFaqID(target.split('=>')[1]);
+			if (!faq) return this.errorReply(`Invalid FAQ.`);
+			const regex = Answerer.stringRegex(target);
+			if (Answerer.queue.includes(target)) {
+				return this.errorReply(`That regex string is already in queue.`);
+			}
+			Answerer.validateRegex(regex);
+			Answerer.queue.push(target);
+			Answerer.writeState();
+			return this.sendReply(`Added "${target}" to the regex suggestion queue.`);
+		},
+		approve(target, room, user) {
+			if (!helpRoom) return this.errorReply(`No Help room to manage the filter for.`);
+			if (!this.can('ban', null, helpRoom)) return false;
+			// intended for use mainly within the page, so supports being used in all rooms
+			this.room = helpRoom;
+			const index = parseInt(target) - 1;
+			if (isNaN(index)) return this.errorReply(`Invalid queue index.`);
+			const str = Answerer.queue[index];
+			if (!str) return this.errorReply(`Item does not exist in queue.`);
+			const regex = Answerer.stringRegex(str);
+			// validated on submission
+			const faq = Answerer.getFaqID(str.split('=>')[1].trim());
+			if (!faq) return this.errorReply(`Invalid FAQ.`);
+			if (!Answerer.data.pairs[faq]) helpData.pairs[faq] = [];
+			Answerer.data.pairs[faq].push(regex);
+			Answerer.queue.splice(index, 1);
+			Answerer.writeState();
+
+			this.privateModAction(`${user.name} approved regex for use with queue number ${target}`);
```

Comment:
I feel like logging the regex itself might also be worthwhile.

Reply:
I mean, it's modlogged already

Reply:
Oh, i misread that